hyperspy/signal.py

```diff
@@ -6550,17 +6550,19 @@ def _plot_permanent_markers(self):
     def add_poissonian_noise(self, keep_dtype=True, random_state=None):
         """Add Poissonian noise to the data.
 
-        This method works in-place. The resulting data type is ``int64``.
-        If this is different from the original data type then a warning
-        is added to the log.
+        This method works in-place. 
 
         Parameters
         ----------
         keep_dtype : bool, default True
+            This parameter is used only for lazy signals. Non-lazy signals
```

Comment:
Dask has been supporting [assignment](https://docs.dask.org/en/stable/array-assignment.html) for a while now - I can't figure out since when from the changelog. Would it be worth using assignment for lazy signal too and simplify/remove this argument (and making a noop with a warning if provided)?

Note for future PR: if we go down that route, our documentation will need updating, for example: https://hyperspy.org/hyperspy-doc/dev/dev_guide/lazy_computations.html#tips-for-writing-methods-that-work-on-lazy-signals

Reply:
Good point, thanks! I have followed your suggestion for `add_poissonian_noise`  and `add_gaussian_noise`